batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/deployment/ArtifactFactoryService.java

```diff
@@ -141,19 +144,34 @@ private BeanManagerImpl getBeanManager() {
         return beanManager == null ? null : BeanManagerProxy.unwrap(beanManager);
     }
 
-    private static Bean<?> getBean(final String ref, final BeanManager beanManager) {
+    private static Bean<?> getBean(final String ref, final BeanManager beanManager, final ClassLoader classLoader) {
         if (beanManager == null) {
```

Comment:
You might consider to add a `|| classLoader == null` because https://github.com/jberet/jsr352/blob/2.1.0.Final/jberet-core/src/main/java/org/jberet/creation/AbstractArtifactFactory.java#L100 would cause a NPE.

Reply:
This method is made a bit complicated by implementing this workaround. Basically what it does is first try to get a CDI bean with a `ref` string name. If no result, then tries to look up a CDI bean with a type that is loaded from the `classLoader`. So if the `classLoader` is null, which will never happen, this method should still perform the first part, which is get the bean by the `ref` name. So no need to check null here.